datafusion/core/src/physical_plan/file_format/row_filter.rs

```diff
@@ -228,11 +230,32 @@ fn size_of_columns(columns: &[usize], metadata: &ParquetMetaData) -> Result<usiz
 /// For a given set of `Column`s required for predicate `Expr` determine whether all
 /// columns are sorted. Sorted columns may be queried more efficiently in the presence of
 /// a PageIndex.
-fn columns_sorted(_columns: &[usize], _metadata: &ParquetMetaData) -> Result<bool> {
-    // TODO How do we know this?
+fn columns_sorted(columns: &[usize], metadata: &ParquetMetaData) -> Result<bool> {
+    // For now we only set sorted with single col with pageIndex
+    if columns.len() == 1 && metadata.page_indexes().is_some() {
+        if let Some(indexes) = metadata.page_indexes().unwrap().get(0) {
+            return match indexes.get(columns[0]).unwrap() {
+                Index::NONE
+                | Index::BOOLEAN(_)
+                | Index::BYTE_ARRAY(_)
+                | Index::FIXED_LEN_BYTE_ARRAY(_) => Ok(false),
```

Comment:
Do we need to support `bool, bytearray, fixed length byte array` future？

we can file a new issue to track this.

Reply:
>Do we need to support bool, bytearray, fixed length byte array in the future？

I think not, in https://github.com/apache/arrow-datafusion/pull/3380, try put columns_sorted col ahead to filter row as much as possible, I think ` bool, bytearray, fixed length byte array` col using `page_index` 's min_max is not an `efficient` filter.

Reply:
Why the filter for `fixed length byte array` is not efficient?
In the parquet, spark will write decimal(20,2) and  encode the value as `fixed length byte array`, we can use the oder to accelerate the filter like `int32,int64`.

Reply:
I am not the expert in the parquet column ColumnOrder, but from the document mentioned above, the order is not the nature order.
The order is determined by logical type not the physical type.